webserver/webserver/src/main/java/io/helidon/webserver/http/RestServer.java

```diff
@@ -0,0 +1,183 @@
+    private RestServer() {
+    }
+
+    /**
+     * Definition of a server endpoint.
+     */
+    @Target(ElementType.TYPE)
+    @Retention(RetentionPolicy.CLASS)
+    @Documented
+    @Inherited
+    @Service.Singleton
+    public @interface Endpoint {
+    }
+
+    /**
+     * Definition of an outbound header (sent with every response).
+     */
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    @Repeatable(Headers.class)
+    @Documented
+    public @interface Header {
+        /**
+         * Name of the header, see {@link io.helidon.http.HeaderNames} constants with {@code _STRING}.
+         *
+         * @return header name
+         */
+        String name();
+
+        /**
+         * Value of the header.
+         *
+         * @return header value
+         */
+        String value();
+    }
+
+    /**
+     * Container for {@link io.helidon.webserver.http.RestServer.Header} repeated annotation.
+     */
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    @Documented
+    public @interface Headers {
+        /**
+         * Headers to add o request.
+         *
+         * @return headers
+         */
+        Header[] value();
+    }
+
+    /**
+     * Definition of an outbound header (sent with every request).
+     */
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    @Documented
+    @Repeatable(ComputedHeaders.class)
+    public @interface ComputedHeader {
+        /**
+         * Name of the header, see {@link io.helidon.http.HeaderNames} constants with {@code _STRING}.
+         *
+         * @return header name
+         */
+        String name();
+
+        /**
+         * A producer type, must be a {@link io.helidon.service.registry.ServiceRegistry} service.
+         *
+         * @return producer to get header value from
+         */
+        Class<? extends HeaderProducer> producerClass();
+    }
+
+    /**
+     * Container for {@link io.helidon.webserver.http.RestServer.ComputedHeader} repeated annotation.
+     */
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    @Documented
+    public @interface ComputedHeaders {
+        /**
+         * Headers to add o request.
+         *
+         * @return headers
+         */
+        ComputedHeader[] value();
+    }
+
+    /**
+     * Header producer, to use with {@link io.helidon.webserver.http.RestServer.ComputedHeader#producerClass()}.
+     */
+    @Service.Contract
+    public interface HeaderProducer {
+        /**
+         * Produce an instance of a named header.
+         *
+         * @param name name to create
+         * @return value for the header
+         */
+        Optional<String> produceHeader(HeaderName name);
+    }
+
+    /**
+     * Listener socket assigned to this endpoint.
+     * This only makes sense for server side, as it is binding endpoint to a server socket.
+     */
+    @Target(ElementType.TYPE)
+    @Retention(RetentionPolicy.CLASS)
+    @Documented
+    @Inherited
+    public @interface Listener {
```

Comment:
I think "socket" should be part of the name. Why not `SocketListener` instead? Seems too generic otherwise.

Reply:
This is aligned with the `Listener` name in Helidon WebServer.